src/libasr/codegen/asr_to_wasm.cpp

```diff
@@ -2376,6 +2808,22 @@ class ASRToWASMVisitor : public ASR::BaseVisitor<ASRToWASMVisitor> {
                                             m_import_func_asr_map["fd_write"])]
                         ->index);
                 wasm::emit_drop(m_code_section, m_al);
+            } else if (t->type == ASR::ttypeType::Complex) {
+                emit_call_fd_write(1, "(", 1, 0);
+                this->visit_expr(*x.m_values[i]);
+                wasm::emit_drop(m_code_section, m_al); // drop imag part
+                if (a_kind == 4) {
+                    wasm::emit_f64_promote_f32(m_code_section, m_al);
+                }
+                wasm::emit_call(m_code_section, m_al, 3  /* print_f64 */);
+                emit_call_fd_write(1, ",", 1, 0);
+                this->visit_expr(*x.m_values[i]);
+                if (a_kind == 4) {
+                    wasm::emit_f64_promote_f32(m_code_section, m_al);
+                }
+                wasm::emit_call(m_code_section, m_al, 3  /* print_f64 */);
+                emit_call_fd_write(1, ")", 1, 0);
+                wasm::emit_drop(m_code_section, m_al); // drop real part
```

Comment:
**Concern:**
**Line 2814 & Line 2826:** When printing a complex number we print the real part and then the imaginary part. It currently happens that, when we fetch a complex number on stack top, the imaginary part of the complex is at the stack top followed by the real part. Thus, we are needing to drop the top stack so that we can print the appropriate real/imag part of the complex number.

Reply:
This concern is now solved/fixed in [WASM: handle_print(): Use globals and fix revisiting expr](https://github.com/lcompilers/lpython/pull/1527/commits/f3d56291abd67b6c207e77fbeaed13dd657077f7).